packages/apollo-server-core/src/runHttpQuery.ts

```diff
@@ -127,16 +129,12 @@ export async function runHttpQuery(
     // debug. Therefore, we need to do some unnatural things, such
     // as use NODE_ENV to determine the debug settings
     return throwHttpGraphQLError(500, [e], {
-      debug: debugFromNodeEnv(process.env.NODE_ENV),
+      debug: debugFromNodeEnv(),
     });
```

Comment:
... when catching the error from `resolveGraphqlOptions`, we need to use whatever is in `process.env.NODE_ENV`.

It looks like it should still be possible to use `request.options.nodeEnv` here, but typescript is failing for me there and I don't understand why.